test/new-e2e/tests/agent-platform/common/agent_behaviour.go

```diff
@@ -30,19 +30,28 @@ func CheckAgentBehaviour(t *testing.T, client *TestClient) {
 
 	t.Run("datadog-agent checks running", func(tt *testing.T) {
 		var statusOutputJSON map[string]any
+		var err error
 		result := false
 		for try := 0; try < 5 && !result; try++ {
-			err := json.Unmarshal([]byte(client.AgentClient.Status(agentclient.WithArgs([]string{"-j"})).Content), &statusOutputJSON)
-			require.NoError(tt, err)
-			if runnerStats, ok := statusOutputJSON["runnerStats"]; ok {
-				runnerStatsMap := runnerStats.(map[string]any)
-				if checks, ok := runnerStatsMap["Checks"]; ok {
-					checksMap := checks.(map[string]any)
-					result = len(checksMap) > 0
+			err = json.Unmarshal([]byte(client.AgentClient.Status(agentclient.WithArgs([]string{"-j"})).Content), &statusOutputJSON)
+			if err == nil {
```

Comment:
nit: you could simplify the code by inverting the check and do something like
```
if err != nil {
    time.Sleep
   continue
}
do the normal processing
```
This would reduce the nested blocks and should allow you to group the checks on `result`

Reply:
nice! on it 🙂